homeassistant/components/awair/const.py

```diff
@@ -0,0 +1,120 @@
+"""Constants for the Awair component."""
+
+from collections import namedtuple
+from datetime import timedelta
+import logging
+
+from homeassistant.const import (
+    ATTR_DEVICE_CLASS,
+    CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+    CONCENTRATION_PARTS_PER_BILLION,
+    CONCENTRATION_PARTS_PER_MILLION,
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_ILLUMINANCE,
+    DEVICE_CLASS_TEMPERATURE,
+    TEMP_CELSIUS,
+    UNIT_PERCENTAGE,
+)
+
+AwairResult = namedtuple("AwairResult", ["device", "air_data"])
+
+API_CO2 = "carbon_dioxide"
+API_DUST = "dust"
+API_HUMID = "humidity"
+API_LUX = "illuminance"
+API_PM10 = "particulate_matter_10"
+API_PM25 = "particulate_matter_2_5"
+API_SCORE = "score"
+API_SPL_A = "sound_pressure_level"
+API_TEMP = "temperature"
+API_TIMEOUT = 20
+API_VOC = "volatile_organic_compounds"
+
+ATTRIBUTION = "Awair air quality sensor"
+
+ATTR_ICON = "icon"
+ATTR_LABEL = "label"
+ATTR_UNIT = "unit"
+
+# Some of these names are historical, because this integration
+# created sensors with names that matched the API somewhat
+# directly. If air quality device classes are standardized
+# across home assistant, then we should adjust these as well.
+# For example:
+# score -> air_quality_index
+# CO2   -> carbon_dioxide
+# PM10  -> particulate_matter_10
+# PM2.5 -> particulate_matter_2_5
+# VOC   -> volatile_organic_compounds
+
+DEVICE_CLASS_AIR_QUALITY_INDEX = "score"
```

Comment:
Only official sensor device classes should be used in the device_class sensor entity property.
https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes

Reply:
Unfortunately that standard didn't exist at the time the original integration was written; or if it did it certainly wasn't required at the time. What you've suggested would be a breaking change for users, and I do not think we should do that - see previous discussion between me and @danielsjf on the PR.

It would actually be worse, because we'd actually be forced to convert this integration from a `sensors` integration into an `air_quality` integration, and for the reasons I outlined in https://github.com/home-assistant/architecture/issues/362 that would be a subpar experience for users compared to what they already have today. Otherwise, adhering to the standard device classes at this time would result in users losing access to most of the data on the devices, rendering the integration relatively useless.

I don't want this PR to force such a breaking change on users, and I believe that we should merge it as-is without forcing it to conform to the [available sensor device classes](https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes) until https://github.com/home-assistant/architecture/issues/362 is resolved.

Reply:
I'm not following about having to convert to an air quality platform. We don't need to do that.

The users would just loose that state attribute nothing more. As it's a static attribute do we think it's used a lot by users?

Reply:
I'm confused too, then - the line for this comment is:

`DEVICE_CLASS_AIR_QUALITY_INDEX = "score"`

We're creating a sensor such as `sensor.living_room_awair_score` - and score is not a static attribute; it changes over the course of the day and is useful on its own.

I read your comment as "do not create this sensor, because it is not one of `{battery,humidity,illuminance,signal_strength,temperature,timestamp,power,pressure}`", but I suspect that isn't what you meant 😆 

The more I think about it, the more I suspect that I fundamentally misunderstand the purpose of a `DEVICE_CLASS`? Is it not needed at all? Looking through the table again I notice that it isn't listed as required...

Reply:
The `device_class` entity property is optional.

I'm not asking to remove all the sensors with custom device class, just that we don't set a custom device class for those entities.

The `device_class` property will be set as an entity state attribute. So not setting the custom device classes will be a breaking change. We could consider keeping it for that reason, but as it's a static attribute maybe the impact is small? Do we think that users are using this state attribute?

Reply:
That makes sense - I did misunderstand what it was used for.

I think that it will be fine to not set non-standard `device_class` properties then, since it's only exposed as a state attribute. I think it's possible someone might be using it that way today, but I would be a little surprised.

I'll update as suggested!